test/testvm.py

```diff
@@ -664,19 +708,29 @@ def _choose_macaddr(self):
                 break
             elif not flavor:
                 macaddrs.append(macaddr)
-        for macaddr in macaddrs:
-            if self._lock_resource(macaddr):
-                return macaddr
-        raise Failure("Couldn't find unused mac address for '%s': %s" % self.flavor, resources)
-
-    def _locate_qemu_kvm(self):
-        rhel_qemu_kvm_path = '/usr/libexec/qemu-kvm'
-        if os.path.exists(rhel_qemu_kvm_path):
-            return rhel_qemu_kvm_path
-        # Assume it's in $PATH
-        return 'qemu-kvm'
-
-    def _start_qemu(self, maintain=False, tty=False, monitor=None):
+        start_time = time.time()
+        # wait a bit before we give up
+        while (time.time() - start_time) < timeout_sec:
```

Comment:
Why are we waiting 30 seconds to get a mac address for a flavor?

Alternate flavors are only used by a single test. Previously we would just fail. It's pretty inherent in our tests that you can't run the same test twice at the same time. Even in this pull request we have statically assigned mac addresses throughout many tests.